webpack/components/AnsibleRoles/components/AnsibleRolesImportButton.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Dropdown, DropdownToggle, DropdownItem } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';
+import { importUrl } from '../AnsibleRolesPageHelpers';
+
+export const AnsibleRolesImportButton = () => {
+  let dropdownItems;
+
+  const {
+    response: { results },
+    status,
+  } = useAPI('get', '/api/v2/smart_proxies');
+
+  const [isOpen, setIsOpen] = React.useState(false);
+  const onToggle = isToggleOpen => {
+    setIsOpen(isToggleOpen);
+  };
+  const onFocus = () => {
+    const element = document.getElementById('toggle-primary');
+    element.focus();
+  };
+  const onSelect = () => {
+    setIsOpen(false);
+    onFocus();
+  };
+
+  if (status !== 'RESOLVED') {
```

Comment:
I'm a fan of explicit error handling instead. Otherwise it keeps "loading" forever.